eclipse-collections/src/main/java/org/eclipse/collections/impl/list/fixed/QuadrupletonList.java

```diff
@@ -145,6 +146,12 @@ public T getOnly()
     }
 
     @Override
+    public Optional<T> getOnlyOptional()
```

Comment:
Code duplication can be reduced by moving `getOptional()` to `AbstractMemoryEfficientMutableList` and overriding it only on empty and singleton